### ISSUE TITLE
Sorting design review for accessibility and design intent compatibility

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#17900]
## Issue Description
Given that a sort-by-date utility does not yet exist in the VSP design system, a mockup has been created to facilitate adding this requested feature to the Find a VA Form product.  The Decision Tools team reviewed several mockups presented by @laurellawrence, believes this design is the most user-friendly and intuitive, and would like to proceed with implementing.

![image](https://user-images.githubusercontent.com/73354907/103291792-89d34080-49ba-11eb-9cd1-aac93fa80f31.png)

https://xd.adobe.com/view/d93ef5ed-5925-444b-a034-26f6739aef5d-8c6d/screen/26ec5437-50c2-4776-b33a-5ecd4786add7/

Prior to code being written, we would like the design to be reviewed by Josh Kim for accessibility recommendations, and also by Ryan Thurlwell, who did the initial design intent, to ensure that it is compatible with his overall design.

---
## Tasks
- [x] Sorting utility mockup review/recommendations by @joshkimux 
- [x] Sorting utility mockup review/approval by @rtwell 

## Acceptance Criteria
- [x] @joshkimux  reviews mockup and provides accessibility recommendations
- [x] @rtwell  reviews mockup and approves based on design intent, or recommends modifications
- [ ] Accessibility and design recommendations are captured in a new issue and included in the the Post-Launch MVP Epic https://github.com/department-of-veterans-affairs/va.gov-team/issues/10758
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by joshkimux** [comment]
From an accessibility perspective, this checks out provided the code is written well. [This article by Huge Giraudel documents a good way of handling it elegantly for screen reader users.](https://dev.edenspiekermann.com/2016/04/14/links-and-accessibility/) It's also been reviewed by Heydon Pickering 👍 

Two, most likely already answered, UX questions for the team and @rtwell :
1. Thinking long term, will more sorts/filters be added on? If so, have we explored more future-proof alternatives?
2. Have alternative, stronger methods of signifying a selection been considered? e.g. adding an icon, background-color, etc.

Otherwise, I think this is clean! 💯

**Comment by MarciMcGuireGCIO** [comment]
Thanks, Josh!  Great info and questions. 

    1. We did discuss future options should a need arise for additional sorting options.  Since this particular request is specific only to an internal VA use case, the relative unobtrusiveness seemed appropriate.  To our knowledge, there is currently no need for a VA or their caregiver to sort the results.

    2. Laurel presented two other designs that were stronger and incorporated background color.  Given the context of the request, however, they felt somewhat heavy-handed.  They would be good candidates if a need for additional sorting arises. Those designs can be seen here:  
    - https://xd.adobe.com/view/d93ef5ed-5925-444b-a034-26f6739aef5d-8c6d/screen/b0086df2-db66-42e9-b202-2f790435e7c3 
    - https://xd.adobe.com/view/d93ef5ed-5925-444b-a034-26f6739aef5d-8c6d/screen/cbec96a1-9e05-4b04-8b5c-961127031fc4  

I hope this is helpful!

**Comment by jenniferlee-dsva** [comment]
Just chiming in - love the simplicity of the sort feature. :-) 

For clarity and historical accuracy - the sorting need isn't for "internal" use case. It is primarily used by non-veteran but external form users such as VSOs and other veteran organizations. But simplicity is always the best design option regardless.  Thank you. 

>Since this particular request is specific only to an internal VA use case...

**Comment by MarciMcGuireGCIO** [comment]
Thank you for the clarification -- I mis-interpreted the audience description as being internal to the VA.

**Comment by rtwell** [comment]
hello lovely people! apologies for not getting eyes on this sooner—i've been on vaca and returned today. @laurellawrence thank you for these comps, particularly for developing mobile versions! 

some notes:
1. I would echo @joshkimux's thoughts about the "select" state. At a glance, this implementation adds some cognitive load when distinguishing the state, as it may be interpreted as a link to another page (esp on mobile). The other versions Laurel provided change the interaction by coupling the sort with the search itself; it looks like in order to sort, the user must click search again which it infers a new search, and loses the current search. that said, the proposed version is so close—If we were to add a something other than an underline to distinguish the select state, i think we'd be there.  
2. as far as future proofing goes, it's always helpful to consider how a pattern like this could be expanded, I would not let that be a blocker. as an agile shop, we should solve for this particular problem, whilst considering scalability as we go, but we can always revisit as needed.

**Comment by MarciMcGuireGCIO** [comment]
Thank you, @rtwell!  Hope you enjoyed your holidays!  

Ryan, @joshkimux -- @1Copenut had a recommendation here --> https://github.com/department-of-veterans-affairs/va.gov-team/issues/17901#issuecomment-753171788 that the "active" sorting state be displayed in black bold text vs.  blue text.  

I also understand the concern with the underline possibly being interpreted as a link.   Forgive my crude mockup (I'm trying to use the equivalent of MS Draw - I miss Create Suite!), and Lauren, but would something in this direction work? 

![image](https://user-images.githubusercontent.com/73354907/103662863-6e2de400-4f3e-11eb-87df-e6f7220c77bf.png)

**Comment by rtwell** [comment]
yes @MarciMcGuireGCIO something like that is definitely an option — I would encourage @laurellawrence to explore!

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO Yes! Icons are great. I'd recommend keeping the blue color on the unselected sort to maintain the visual affordance that it's interactive. Some sighted users may misinterpret it as being inactive otherwise.

**Comment by laurellawrence** [comment]
Based on the feedback from @joshkimux and @rtwell, a few more iterations on this can be found here: https://xd.adobe.com/view/d5e7301d-12f9-438f-b30e-3646e927ed70-80b4/?hints=off

**Comment by rtwell** [comment]
This design doesn't look interactive to me because of the lack of color. however, displaying a select state may also contribute to some of the confusion b/c now we need another signifier to say "select" **and** a "not-select" state. I'm wondering if a drop down pattern would be useful because it only needs to show sort type at a time. @joshkimux i defer to your expertise regarding how a11y-friendly this approach is.

here's an example from github (just an example, not the _perfect_ example):

![image](https://user-images.githubusercontent.com/12564977/103695268-43598500-4f6a-11eb-845d-155a7dc49e8d.png)

**Comment by laurellawrence** [comment]
@rtwell there is a sort/filter dropdown example in our sketch library similar to this but it is not in the online va.gov design system. There were some accessibility concerns with this since there wasn't an explicit button press for sort after the dropdown menu selection. The benefit of a dropdown does give us flexibility of adding other filters in the future but the scope of this work only states that we need to filter by last updated. Here is the first set of iterations that includes this example provided in our sketch library.  https://xd.adobe.com/view/d93ef5ed-5925-444b-a034-26f6739aef5d-8c6d/screen/572321bc-2c7c-48fa-aba5-746f7300e8c7

**Comment by rtwell** [comment]
• if we are sorting by "last updated", then we should be explicit about that in the label
• because something is not in the design system does not mean we cannot add to it
• the up and down carats are used throughout va.gov for opening menus, and we should be consistent across the site in how things behave— i think you could use the [sort icon](https://fontawesome.com/icons/sort?style=solid) in lieu.

**Comment by joshkimux** [comment]
Hey y'all, sorry for being a bit late to the party. Here are my thoughts:

I agree with @rtwell that it should be explicit what item has been selected. Ideally in a written format as opposed to strictly visual. Also agree with @laurellawrence that it's good to stick to the scope of work, a select box for only two options may be going too far in terms of how much weight it adds to the page. And, I agree with both of y'all that we should focus on consistency _and_ avoid styling this to look too closely to a link.

What are your thoughts on approaching it like this?:
![image](https://user-images.githubusercontent.com/14154792/103700979-43aa4e00-4f73-11eb-9044-7e6e5a112439.png)

This accomplishes a couple of things:
- Styling it as a button is materially honest, as these are buttons
- This would provide an explicit button press to the user
- Larger touch target for mobile users compared to just a link
- Not a select box
- Adding the "sorted by" into the "showing" string accomplishes @rtwell 's point of making the sort selection explicit 

On the flip side, this probably does not exists in the design system and could add unnecessary weight. In which case, I think a select box may be more appropriate.

<img width="251" alt="Screen Shot 2021-01-05 at 4 00 10 PM" src="https://user-images.githubusercontent.com/14154792/103701197-984dc900-4f73-11eb-91f5-963b493d0e59.png">
<img width="274" alt="Screen Shot 2021-01-05 at 4 00 14 PM" src="https://user-images.githubusercontent.com/14154792/103701200-997ef600-4f73-11eb-96e5-f8d23ef8036a.png">

To address @laurellawrence 's points, I see the concern over there not being an explicit button press. I think it's alright in this instance because it's a relatively frequent pattern, and states can be announced through aria usage.

<img width="1132" alt="Screen Shot 2021-01-05 at 4 37 47 PM" src="https://user-images.githubusercontent.com/14154792/103701648-64bf6e80-4f74-11eb-8209-eb88c118505a.png">


Let me know y'alls thoughts!
